service/lib/dinstaller/dbus/question.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2022] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "dbus"
+require "pathname"
+require "dinstaller/dbus/questions"
+require "dinstaller/question"
+require "dinstaller/luks_question"
+
+module DInstaller
+  module DBus
+    # Class to represent a question on D-Bus
+    #
+    # Questions are dynamically exported on D-Bus. All questions are exported as children of
+    # {DBus::Questions} object to mimic ObjectManager behavior. Note that ruby-dbus does not support
+    # ObjectManager yet.
+    #
+    # Clients should provide an answer for each question.
+    class Question < ::DBus::Object
+      # This module contains the interfaces a question can implement. The interfaces that a question
+      # implements are dynamically determined while creating a new question.
```

Comment:
I think you should explicitelly mention that any class that use this modules has to be ::DBus::Object otherwise it failed when calling dbus interface.